oxidation/crates/parsec_api_protocol/src/error.rs

```diff
@@ -0,0 +1,49 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BSLv1.1 (eventually AGPLv3) 2016-2021 Scille SAS
+
+use parsec_api_types::DateTime;
+use thiserror::Error;
+
+use crate::ApiVersion;
+
+#[derive(Error, Debug, PartialEq)]
+pub enum HandshakeError {
+    #[error("Failed chalenge {0:?}")]
+    FailedChallenge(String),
```

Comment:
I guess we can also remove the `String` field here too, and just have a `"Handshake invalid answer (bad signature or challenge value)"` constant error message.

Reply:
Done :+1: